common/src/uint/impls.rs

```diff
@@ -1027,4 +1076,13 @@ mod tests {
         test_op_equality(10000000000000000000000000);
         test_op_equality(100000000000000000000000000);
     }
+
+    #[test]
+    pub fn uint512_from_uint256() {
```

Comment:
Please add more aggressive testing. See how I tested 128 vs 256.

Reply:
I rewrote this test and an old one using proptest. Please take a look